01 Operator Overloading/OperatorOverloading.Host/Transaction.cs

```diff
@@ -0,0 +1,125 @@
+
+
+                while (true) //this while keep on taking the input unless you enter valid amount.
+                {
+                    Console.WriteLine("Enter amount for that Currency.");
+                    
+                        string input = Console.ReadLine();
+                        double temp = 0.0;
+                        if (Double.TryParse(input, out temp))
+                        {
+                            money1.Amount = temp;
+                            break;
+                        }
+                        else
+                        {
+                            Console.WriteLine("Invalid Amount, Enter again: ");
+                            continue;
+                        }
+                   
+                }
+   
+             Console.WriteLine();
+
+             while (true)// condition to check currency is not empty
+             {
+                 Console.WriteLine("Enter 2nd Currency.");
+                 money2.Currency = Console.ReadLine();
+                 if (money2.Currency.Equals(""))
+                 {
+                     Console.WriteLine("Invalid Currency.");
+                     continue;
+                 }
+                 else
+                     break;
+
+             }
+
+            
+            while (true) //this while keep on taking the input unless you enter valid amount.
+            {
+                Console.WriteLine("Enter amount for that Currency.");
+
+                string input = Console.ReadLine();
+                double temp = 0.0;
+                if (Double.TryParse(input, out temp))
+                {
+                    money2.Amount = temp;
+                    break;
+                }
+                else
+                {
+                    Console.WriteLine("Invalid Amount, Enter again: ");
+                    continue;
+                }
+                   
+            }
+
+            try
+            {
+                money3 = money1 + money2;
+            }
+            /* this catch handles the invalid currency exception which is defined by the user.
+             * out of range exception occurs when the currencies of both the adding money is different*/
+            catch (InvalidCurrencyException exception) //this catch handles the Invalid Currency Exception which is defined by the user.
+            {
+                Console.WriteLine("" + exception.Message);
+                Console.ReadKey();
+                return;
+            }
+            /* this catch handles the Out of range exception which is defined by the user.
+             * out of range exception occurs when range of double is exceeded after addition*/
+            catch (OutOfRangeException exception)
+            {
+                Console.WriteLine("" + exception.Message);
+                Console.ReadKey();
+                return;
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("" + e.StackTrace);
```

Comment:
Why are you concatenating empty string before exception message.
And if you are always going to print the message you don't need to catch three different exceptions. Just catch System.Exception